### PR TITLE
Fix optional imports

### PR DESCRIPTION
**Pull request by theengineear**
This is a stand-alone module to do the following:

* Don’t import optional things until we need them
* Keep all the optional imports centralized

### CODE REVIEW

**Comment by theengineear** [comment]
Sorry for all the review requests guys. This one should be reasonably simple as well though. It's just a cleanup of how we figure out if optional imports have been imported. One major motivation for me was to stop importing `tools` *everywhere*. As usual, it's a constant source of cyclic imports, so factoring out these requirements is great.

**Comment by theengineear** [comment]
@chriddyp @Kully ^^

File: plotly/figure_factory/_annotated_heatmap.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-from plotly import exceptions
+from plotly import exceptions, optional_imports
 from plotly.figure_factory import utils
 from plotly.graph_objs import graph_objs
-from plotly.tools import _numpy_imported
 
-if _numpy_imported:
-    import numpy as np
+# Optional imports, may be None for users that only use our core functionality.
+np = optional_imports.get_module('numpy')
```
Comment (chriddyp) [inline plotly/figure_factory/_annotated_heatmap.py]: fresh and clean! 🛀

**Comment by chriddyp** [comment]
very nice! 💃

**Comment by theengineear** [comment]
@Kully gonna merge so I can start doing some manual testing on `2.0.0`.

**Comment by Kully** [comment]
> @Kully gonna merge so I can start doing some manual testing on 2.0.0.

You wanted me to review correct?

**Comment by theengineear** [comment]
@Kully yah, that'd be great, but I'm sure I'll be asking for manual testing / review help before I attempt to merge `2.0.0` anyhow. I.e., if you don't have time to review this one, that's OK.

**Pull request merged**